LLama/LLamaWeights.cs

```diff
@@ -204,11 +213,33 @@ public static async Task<LLamaWeights> LoadFromFileAsync(IModelParams @params, C
                 return model;
             }
         }
+        #endregion
 
         /// <inheritdoc />
         public void Dispose()
         {
-            NativeHandle.Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Unload all models when called explicitly via dispose
+        /// </summary>
+        /// <param name="disposing">Whether or not this call is made explicitly(true) or via GC</param>
+        internal void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                NativeHandle.DangerousRelease();
```

Comment:
I think `DangerousRelease` needs to happen, whether or not this being called by the finalizer.

Example:
1. Allocate a SafeHandle
2. Wrap in LLamaWeights (increment count)
3. Drop LLamaWeights without properly calling dispose
4. `DangerousRelease` never gets called, so refcount never goes to zero!

Reply:
updated